### PR TITLE
Remove unused custom excertion handling in all KeyVault

### PR DESCRIPTION
**Pull request by lmazuel**
It feels KeyVault has a custom exception handling that is:
- Incorrect Typing. Mypy yells if I improve azure-core typing (it's how I ended up here in the first place)
- Useless. It's mapping to type in Azure-Core, which is already the behavior of azure-core.
- Masking features. The way it was done, would prevent customers to pass `error_map`. I believe we wanted `error_map` to be accessible to customers (to be verified, I'm not sure on this point)

I remove all that code, didn't change any tests ........ and everything is green :). The file was not even plugged in the `a-kv-admin` package.

So either this file was indeed useless, or we miss some tests on error scenarios ;)

### CODE REVIEW

**Comment by lmazuel** [comment]
More context after discussion with @johanste :
- We believe this code was to solve https://github.com/Azure/azure-sdk-for-python/issues/9690
- The problem solved by this issue, has been removed by recent codegen: we don't generate custom exception anymore
- You can check for instance that [this link](https://github.com/Azure/azure-sdk-for-python/blob/83a68333b9be0a1e6b46f92b625b4c6bf801bd03/sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_generated/v7_0/operations/_key_vault_client_operations.py#L4492) do not mention a `KeyVaultErrorException` anymore

**Review by mccoyp — APPROVED** [review]

**Comment by mccoyp** [comment]
@johanste @annatisch Are there any concerns from either of you about the exception handling in Key Vault? Laurent's fix looks good to me (and is tremendously appreciated) 🙂

**Comment by johanste** [comment]
I think all the concerns I had have been addressed.

File: sdk/keyvault/azure-keyvault-certificates/assets.json [hunk 1]
```diff
@@ -2,5 +2,5 @@
   "AssetsRepo": "Azure/azure-sdk-assets",
   "AssetsRepoPrefixPath": "python",
   "TagPrefix": "python/keyvault/azure-keyvault-certificates",
-  "Tag": "python/keyvault/azure-keyvault-certificates_e24380c294"
+  "Tag": "python/keyvault/azure-keyvault-certificates_f472924e90"
```
Comment (scbedd) [inline sdk/keyvault/azure-keyvault-certificates/assets.json:5]: Comment for location clarity.

**Pull request merged**